src/pkg/oidc/helper.go

```diff
@@ -356,6 +356,11 @@ func userInfoFromClaims(c claimsProvider, setting cfgModels.OIDCSetting) (*UserI
 
 		if username, ok := allClaims[setting.UserClaim].(string); ok {
 			res.autoOnboardUsername = username
+			// If User is not onboarded yet, store Username as well
+			// Otherwise, onboard flow would fail if "name" is not set in token
+			if res.Username == "" {
```

Comment:
This `if` may not be needed, b/c at this point the Username may be set if the token has `name` attribute, but the expected result is that the Username is always set as the autoOnboard setting.

Could you please also add a comment to explicitly explain that the func `mergeUserInfo` may not be called, so the `res.Username` must be set in this case.

Reply:
I've removed the conditional and added some extra comments regarding the purpose of this change.